File: Hosts/Silverlight/Microsoft.Scripting.Silverlight/repl_formatter.rb
include Microsoft::Scripting::Silverlight

class RubyReplFormatter
  include IReplFormatter

  def initialize(repl, formatter)
    @repl = repl
    @formatter = formatter
  end

  def prompt_element(element)
  end

  def prompt_html
  end

  def sub_prompt_html
  end

  def format(obj)
    "=> #{obj.inspect}"
  end
end

def create_repl_formatter(repl, formatter)
  RubyReplFormatter.new(repl, formatter)
end

File: Hosts/Silverlight/Microsoft.Scripting.Silverlight/repl_formatter.py
from Microsoft.Scripting.Silverlight import Repl, IReplFormatter

class PythonReplFormatter(IReplFormatter):
    def __init__(self, repl, formatter):
        self.defaultFormatter = formatter

    def PromptElement(self, element):
        element.SetStyleAttribute("color", "yellow")
   
    def PromptHtml(self):
        pass

    def SubPromptHtml(self):
        pass

    def Format(self, obj):
        if obj is None:
            return None
        return "%s" % repr(obj)

def create_repl_formatter(repl, formatter):
    return PythonReplFormatter(repl, formatter)
